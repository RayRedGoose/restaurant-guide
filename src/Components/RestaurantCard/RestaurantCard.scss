.restaurant-card {
  border: 1px solid black;
  display: flex;
  justify-content: space-between;
  margin: 2rem auto;
  padding: 1rem;
  width: 85%;
  section {
    padding: 1rem;
  }
}

.card-description {
  h2 {
    font-size: 1.5rem;
    font-weight: 700;
    margin-bottom: 1rem;
  }
  main {
    display: flex;
    flex-wrap: wrap;
    p {
      margin-right: 1.25rem;
    }
  }
}

.card-information {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 40%;
  p {
    display: flex;
    align-items: center;
    line-height: 1.4;
    position: relative;
    &::before {
      content: '';
      background-position: center;
      background-repeat: no-repeat;
      background-size: 100%;
      min-width: 18px; height: 18px;
      margin-right: 0.5rem;
    }
    &.location {
      width: 50%;
      &::before {
        background-image: url('../../assets/images/pin.svg');
      }
    }
    &.phone {
      width: 40%;
      &::before {
        background-image: url('../../assets/images/phone.svg');
      }
    }
  }
}
