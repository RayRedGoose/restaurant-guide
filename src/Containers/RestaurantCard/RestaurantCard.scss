@import "../../assets/css/variables.scss";

.restaurant-card {
  background-color: #fff;
  border-radius: 1rem;
  border-bottom: 5px solid $box-color;
  box-shadow: 0px 0px 12px rgba(23, 126, 137, 0.2);
  margin: 1rem auto;
  padding: 2rem 2rem 0;
  width: 70%;
  header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    section * {
      display: inline-block;
      margin-right: 0.5em;
      background-color: $label-color;
      border-radius: 1em;
      color: $dark-color;
      font-size: 0.75rem;
      font-weight: 700;
      letter-spacing: 0.03em;
      padding: 0.4rem 0.8rem;
    }
  }
}

footer {
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0.75rem 2rem;
  margin: 1rem -2rem 0;
  font-size: 0.85rem;
  font-weight: 700;
  svg {
    margin-left: 0.5rem;
  }
  &:focus,
  &:hover {
    background-color: rgba(23, 126, 137, 0.05);
    outline: none;
  }
}
.card-description {
  margin: 1rem 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  h2 {
    grid-area: title;
    width: 40%;
    a {
      border-bottom: 2px solid transparent;
      color: $accent-color;
      font-family: $accent-font;
      font-size: 2rem;
      font-weight: 700;
      letter-spacing: 0.05em;
      line-height: 1.25em;
      transition: border-bottom 0.5s;
      text-decoration: none;
      &:focus,
      &:hover {
        border-bottom: 2px solid $accent-color;
        outline: none;
        & ~ svg {
          opacity: 1;
        }
      }
    }
    svg {
      opacity: 0;
      height: 1.5rem;
      margin-left: 0.5rem;
      transition: opacity 0.5s;
    }
  }
  h4 {
    display: block;
    font-weight: 700;
    margin: 2rem 0 1rem;
  }
  .location,
  .open-hour {
    p {
      display: block;
      margin-bottom: 0.75rem;
    }
  }
  .location {
    grid-area: map;
  }
  .open-hour {
    grid-area: hours;
    h4 {
      margin-top: 1rem;
    }
  }
  &.opened {
    display: grid;
    grid-template-areas: "title map" "hours map";
    grid-template-columns: 1.75fr 1.25fr;
    grid-template-rows: 1fr 2fr;
    * {
      align-self: start;
    }
    h2 {
      width: 80%;
    }
  }
}

.tags {
  display: flex;
  p {
    // font-size: 0.85rem;
    font-weight: 700;
    color: $accent-color;
    margin-right: 0.5em;
  }
}

.additional-info {
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: $accent-color;
  font-size: 0.85rem;
  font-weight: 700;
  text-transform: capitalize;
  margin-top: 2rem;
  .attires {
    * {
      display: inline-block;
    }
    svg {
      margin-right: 0.5rem;
    }
  }
}

.info {
  display: flex;
  justify-content: flex-end;
  min-width: 50%;
  p {
    margin-left: 2rem;
  }
  svg {
    margin-right: 0.5rem;
  }
}
