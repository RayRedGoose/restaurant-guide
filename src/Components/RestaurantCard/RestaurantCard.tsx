import './RestaurantCard.scss';
import React from 'react';
import { IRestaurantObject } from 'assets/ts/interfaces';

interface Props {
  restaurant: IRestaurantObject
}

const RestaurantCard: React.FC<Props> = ({ restaurant }) => {
  const { name, city, state, telephone, genre } = restaurant;
  const cityNew = city.replace(/,/g, ', ');
  const genreBlocks: JSX.Element[] = genre
    .split(',')
    .map((gnr: string, ind: number) => (
      <p key={gnr + ind} className='genre'>{gnr}</p>
    ));

  return (
    <section className="restaurant-card">
      <section className="card-description">
        <h2>{name}</h2>
        <main>
          {genreBlocks}
        </main>
      </section>
      <section className="card-information">
        <p className="location">{`${cityNew}, ${state}`}</p>
        <p className="phone">{telephone}</p>
      </section>
    </section>
  );
};

export default RestaurantCard;
