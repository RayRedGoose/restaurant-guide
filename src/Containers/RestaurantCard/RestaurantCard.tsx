import "./RestaurantCard.scss";
import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faAngleDown,
  faAngleUp,
  faAngleDoubleRight,
  faBriefcase,
  faGlasses,
  faUserTie,
  faTshirt,
  faMapMarkerAlt,
  faPhoneAlt,
} from "@fortawesome/free-solid-svg-icons";

import { IRestaurantObject, IAppStore } from "assets/ts/interfaces";
import Map, { Props as MapProps } from "Components/Map/Map";

interface Props {
  restaurant: IRestaurantObject;
}

const RestaurantCard: React.FC<Props> = ({ restaurant }) => {
  const [isClicked, setClicked] = useState<boolean>(false);
  const {
    name,
    city,
    state,
    telephone,
    genre,
    hours,
    attire,
    address1,
    lat,
    long,
    zip,
    tags,
    website,
  } = restaurant;
  const store = useSelector((store: IAppStore) => store);

  useEffect(() => {
    setClicked(false);
  }, [store]);

  const toggleCard = (): void => setClicked(!isClicked);

  const cityNew = city.replace(/,/g, ", ");

  const mapProps: MapProps = {
    name,
    longitude: parseFloat(long),
    latitude: parseFloat(lat),
  };

  const genreBlocks: JSX.Element[] = genre
    .split(",")
    .map((gnr: string, ind: number) => <p key={gnr + ind}>{gnr}</p>);

  const openHours: JSX.Element[] = hours
    .split("; ")
    .map((hour: string, ind: number) => <p key={`hour${ind}`}>{hour}</p>);

  const tagLines: JSX.Element[] = tags
    .split(",")
    .reduce((acc: string[], tag: string) => {
      if (!acc.includes(tag)) acc = [...acc, tag];
      return acc;
    }, [])
    .map((tag: string, ind: number) => <p key={`tag${ind}`}>#{tag}</p>);

  const AttireLabel = (): JSX.Element => (
    <p className="attires">
      {attire.includes("business") && <FontAwesomeIcon icon={faBriefcase} />}
      {attire.includes("smart") && <FontAwesomeIcon icon={faGlasses} />}
      {attire === "formal" && <FontAwesomeIcon icon={faUserTie} />}
      {attire === "casual" && <FontAwesomeIcon icon={faTshirt} />}
      <p className="label">{attire}</p>
    </p>
  );

  const Location = (): JSX.Element => (
    <div className="location">
      <Map {...mapProps} />
      <h4>Address:</h4>
      <p>{address1}</p>
      <p>{`${cityNew}, ${state} ${zip}`}</p>
    </div>
  );

  return (
    <section className="restaurant-card" tabIndex={0}>
      <header>
        <section>{genreBlocks}</section>
      </header>
      <section
        className={isClicked ? "card-description opened" : "card-description"}
      >
        <h2>
          <a href={website} target="_blank" rel="noopener noreferrer">
            {name}
          </a>
          <FontAwesomeIcon icon={faAngleDoubleRight} />
        </h2>
        {!isClicked ? (
          <main className="info">
            <p>
              <FontAwesomeIcon icon={faMapMarkerAlt} />
              {`${cityNew}, ${state}`}
            </p>
            <p>
              <FontAwesomeIcon icon={faPhoneAlt} />
              {telephone}
            </p>
          </main>
        ) : (
          <main className="open-hour">
            <h4>Open:</h4>
            {openHours}
          </main>
        )}
        {isClicked && <Location />}
      </section>
      {isClicked && (
        <section className="additional-info">
          <p className="tags">{tagLines}</p>
          <AttireLabel />
        </section>
      )}
      <footer tabIndex={0} onClick={toggleCard} onKeyPress={toggleCard}>
        <p>{!isClicked ? "More" : "Less"}</p>
        <FontAwesomeIcon icon={!isClicked ? faAngleDown : faAngleUp} />
      </footer>
    </section>
  );
};

export default RestaurantCard;
