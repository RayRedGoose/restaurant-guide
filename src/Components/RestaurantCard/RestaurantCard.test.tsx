import React from 'react';
import { shallow } from 'enzyme';
import RestaurantCard from './RestaurantCard';

it('should match the snapshot', () => {
  const mockRestaurant = {
    id: 'ac11',
    name: 'Old Hickory Steakhouse',
    address1: "201 Waterfront St",
    city: "Oxon Hill",
    state: "MD",
    zip: "20745",
    lat: "38.782098",
    long: "-77.017492",
    telephone: "(301) 965-4000",
    tags: "Social,Food and Dining,Restaurants,Steakhouses",
    website: "http://www.gaylordnational.com",
    genre: "Steak,American,Contemporary,Seafood,Cafe",
    hours: "Open Daily 5:30 PM-10:00 PM",
    attire: "business casual"
  };
  
  const wrapper = shallow(
    <RestaurantCard restaurant={mockRestaurant} />
  );
  expect(wrapper).toMatchSnapshot();
});
