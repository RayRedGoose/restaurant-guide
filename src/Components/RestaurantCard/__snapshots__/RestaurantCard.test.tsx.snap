// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section
  className="restaurant-card"
>
  <section
    className="card-description"
  >
    <h2>
      Old Hickory Steakhouse
    </h2>
    <main>
      <p
        className="genre"
        key="Steak0"
      >
        Steak
      </p>
      <p
        className="genre"
        key="American1"
      >
        American
      </p>
      <p
        className="genre"
        key="Contemporary2"
      >
        Contemporary
      </p>
      <p
        className="genre"
        key="Seafood3"
      >
        Seafood
      </p>
      <p
        className="genre"
        key="Cafe4"
      >
        Cafe
      </p>
    </main>
  </section>
  <section
    className="card-information"
  >
    <p
      className="location"
    >
      Oxon Hill, MD
    </p>
    <p
      className="phone"
    >
      (301) 965-4000
    </p>
  </section>
</section>
`;
